deployment/ccip/changeset/testhelpers/test_environment.go

```diff
@@ -316,6 +324,55 @@ func (m *MemoryEnvironment) MockUSDCAttestationServer(t *testing.T, isUSDCAttest
 	return endpoint
 }
 
+// mineBlocks forces the simulated backend to produce a new block every X seconds
+// If you need to manually commit blocks, you must use the returned commit func, rather than calling Commit() directly,
+// which will race.
```

Comment:
Just wanted to point out that this race is already covered within `memory.Backend` itself, see 

```
func (b *Backend) Commit() common.Hash {
	b.mu.Lock()
	defer b.mu.Unlock()
	return b.Sim.Commit()
}
```

not sure if this helps you simplify further or not

Reply:
Good point, I directly copied this from cltest to avoid the dependency but we can simplify